src/plugins/irc-events/banlist.js

```diff
@@ -6,19 +6,36 @@ const Msg = require("../../models/msg");
 module.exports = function(irc, network) {
 	const client = this;
 
-	irc.on("banlist", function(banlist) {
-		const channel = banlist.channel;
-		const bans = banlist.bans;
+	irc.on("banlist", (list) => {
+		const data = list.bans.map((ban) => ({
+			hostmask: ban.banned,
+			banned_by: ban.banned_by,
+			banned_at: ban.banned_at * 1000,
+		}));
+
+		handleList(Chan.SpecialType.BANLIST, "Ban list", list.channel, data);
+	});
+
+	irc.on("inviteList", (list) => {
+		const data = list.invites.map((invite) => ({
+			hostmask: invite.invited,
+			invited_by: invite.invited_by,
+			invited_at: invite.invited_at * 1000,
+		}));
 
-		if (!bans || bans.length === 0) {
+		handleList(Chan.SpecialType.INVITELIST, "Invite list", list.channel, data);
+	});
+
+	function handleList(type, name, channel, data) {
```

Comment:
Do we need type *and* name? We could remove name, I think, and just reformat the type for the UI. They are a consistent format, after all.

Reply:
Ideally it would be entirely in the UI yeah, but we don't have a translation framework in place yet.

There's also the error message saying X list is empty.

Reply:
Yeah but we can just take off the end of the type and use that for the error message.

Reply:
What do you mean?